Add workspaces to skipgram/cbow

agibsonccc opened the pull request:
Add new unified profiler for reporting op execution and memory
Add new workspace reporting to allocations tracker
 Remove extra workspace declaration for scalar value, allow it to fall within current workspace Allow stack aggregators to return strings from their renderTree calls for printing
 Migrate printing of op profiler op dashboard to a string we can print wherever we want 
Misc code clean up (delete commented code etc)
Spacing

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

agibsonccc committed:
Add workspaces to skipgram/cbow
Add new unified profiler for reporting op execution and memory
Add new workspace reporting to allocations tracker
Remove extra workspace declaration for scalar value, allow it to fall within current workspace
Allow stack aggregators to return strings from their renderTree calls for printing
Migrate printing of op profiler op dashboard to a string we can print wherever we want
Misc code clean up (delete commented code etc)
Spacing

agibsonccc committed:
Add workspaces to skipgram/cbow
Add new unified profiler for reporting op execution and memory
Add new workspace reporting to allocations tracker
Remove extra workspace declaration for scalar value, allow it to fall within current workspace
Allow stack aggregators to return strings from their renderTree calls for printing
Migrate printing of op profiler op dashboard to a string we can print wherever we want
Misc code clean up (delete commented code etc)
Spacing

agibsonccc committed:
Add workspaces to skipgram/cbow
Add new unified profiler for reporting op execution and memory
Add new workspace reporting to allocations tracker
Remove extra workspace declaration for scalar value, allow it to fall within current workspace
Allow stack aggregators to return strings from their renderTree calls for printing
Migrate printing of op profiler op dashboard to a string we can print wherever we want
Misc code clean up (delete commented code etc)
Spacing
Add javacpp off heap/heap memory

agibsonccc committed:
Add workspaces to skipgram/cbow
Add new unified profiler for reporting op execution and memory
Add new workspace reporting to allocations tracker
Remove extra workspace declaration for scalar value, allow it to fall within current workspace
Allow stack aggregators to return strings from their renderTree calls for printing
Migrate printing of op profiler op dashboard to a string we can print wherever we want
Misc code clean up (delete commented code etc)
Spacing
Add javacpp off heap/heap memory

treo commented on libnd4j/include/legacy/cpu/NativeOpExecutioner.cpp:
```diff
@@ -829,13 +829,14 @@ void NativeOpExecutioner::execScalar(sd::LaunchContext *lc, int opNum, const voi
   BUILD_PAIRWISE_SELECTOR(xType, yType, zType, functions::scalar::ScalarTransform,
                           ::transform(opNum, hX, hXShapeInfo, hZ, hZShapeInfo, hScalar, extraParams), SD_COMMON_TYPES,
                           SD_COMMON_TYPES_ALL);
+  sd_printf("execScalar: RAfter unning experimental transform\n",0);
```
Should this really be an `sd_printf`? Even if someone builds with the experimental flag, I'd assume that this would be meant to be a debug print?

agibsonccc replied:
Fixed. The editor doesn't highlight parts that don't show up in the ifdef..just missed that one. Thanks for highlighting!

agibsonccc committed:
Fix cbow inference segfaults
Fix scalar definition
Misc spacing fixes
Remove commented code
Remove extra print
